### DISCUSSION TITLE
In-game screen frequency change

### DISCUSSION THREAD

**Discussion by Dopaminos** [ppy/osu#18732] [Q&A] [answered]
Hello, i have a laptop with R7 5800H + 3050ti 144hz screen running on Manjaro Linux with second 60hz screen.
I use mesa drivers + nvidia from pacman. For some reason, game shows that it's 144hz, but it feels like clear 60hz.
Meanwhile on Windows 11 all is ok. 
To make my screen 144 + make 2nd work i use 
optimus-manager --switch nvidia
xrandr --newmode "1920x1080_144.00"  452.50  1920 2088 2296 2672  1080 1083 1088 1177 -hsync +vsync && xrandr --addmode eDP-1-1 1920x1080_144.00 && xrandr --output eDP-1-1 --mode 1920x1080_144.00

what could be a source of problem?
logs:
[network.log](https://github.com/ppy/osu/files/8917280/network.log)
[performance.log](https://github.com/ppy/osu/files/8917281/performance.log)
[runtime.log](https://github.com/ppy/osu/files/8917282/runtime.log)
[database.log](https://github.com/ppy/osu/files/8917279/database.log)

making game fullscreen, borderless or windowed doesn't change anything about it

**Comment by smoogipoo** [comment]
What does your monitor layout look like? Does disconnecting the 60Hz screen fix it?

**Reply by Dopaminos** [reply]
![image](https://user-images.githubusercontent.com/42616718/174255282-f63f5850-bdf3-4c78-9a8c-8456c0f1b09b.png)

> Does disconnecting the 60Hz screen fix it?

Nah, it's not.

**Reply by Dopaminos** [reply]
seems there is no solution. sad

**Comment by smoogipoo** [comment]
Does disabling the compositor help?

**Reply by Dopaminos** [reply]
Nope :(

**Reply by Dopaminos** [reply]
disabled it with ``qdbus org.kde.KWin /Compositor org.kde.kwin.Compositing.suspend``

**Comment by smoogipoo** [comment]
Can you try running `glxgears` (you might need `mesa-utils`) for a while and seeing what it outputs to the CLI? Should be something like this:
```
$ glxgears
Running synchronized to the vertical refresh.  The framerate should be
approximately the same as the monitor refresh rate.
750 frames in 5.0 seconds = 149.794 FPS
720 frames in 5.0 seconds = 143.853 FPS
720 frames in 5.0 seconds = 143.849 FPS
720 frames in 5.0 seconds = 143.859 FPS
```

**Reply by Dopaminos** [reply]
Without switching with optimus it's pretty okay but after switch to nvidia...
```
Running synchronized to the vertical refresh.  The framerate should be
approximately the same as the monitor refresh rate.
54207 frames in 5.0 seconds = 10841.358 FPS
60032 frames in 5.0 seconds = 12006.306 FPS
64585 frames in 5.0 seconds = 12916.700 FPS
65408 frames in 5.0 seconds = 13081.523 FPS
```

**Reply by smoogipoo** [reply]
No clue unfortunately. Your configuration looks borked from that output.

**Comment by Dopaminos** [comment, marked as answer]
@smoogipoo i found an answer and it's pretty simple.
![изображение](https://user-images.githubusercontent.com/42616718/186746996-143e7f02-ed32-4cab-932d-4e0edd1a682d.png)
after reboot, without an optimus-manager all was working ok. :(

**Reply by Dopaminos** [reply]
xorg renders all by minimal refreshing rate of screens

**Reply by Dopaminos** [reply]
it didn't worked here https://github.com/ppy/osu/discussions/18732#discussioncomment-2969649
cause i didn't rebooted system after disabling screen
